packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3338,13 +3338,16 @@ export class ContainerRuntime
 		// Make sure all BatchManagers are empty before entering staging mode,
 		// since we mark whole batches as "staged" or not to indicate whether to submit them.
 		this.outbox.flush();
+		this.channelCollection.notifyStagingMode(true);
```

Comment:
Maybe put this at the end of the function? So if someone checks `inStagingMode` it matches the paramerer's value